### PR TITLE
Add 2.11 release note

### PR DESCRIPTION
**Pull request by heemin32**
### Description
Add 2.11 release note

### Issues Resolved
https://github.com/opensearch-project/neural-search/issues/317

### Check List
- [X] New functionality includes testing.
    - [X] All tests pass
- [X] New functionality has been documented.
    - [X] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by vibrantvarun — APPROVED** [review]

**Review by martin-gaievski — APPROVED** [review]

File: release-notes/opensearch-neural-search.release-notes-2.11.0.0.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## Version 2.11.0.0 Release Notes
+
+Compatible with OpenSearch 2.11.0
+
+### Features
```
Comment (navneet1v) [inline release-notes/opensearch-neural-search.release-notes-2.11.0.0.md:5]: why multi model feature is not added in release notes?

**Comment by navneet1v** [comment]
@martin-gaievski

File: release-notes/opensearch-neural-search.release-notes-2.11.0.0.md (hunk 1 above)
Comment (navneet1v) [reply release-notes/opensearch-neural-search.release-notes-2.11.0.0.md:5]: @martin-gaievski

Comment (heemin32) [reply release-notes/opensearch-neural-search.release-notes-2.11.0.0.md:5]: it went to directly to 2.x branch. I might need to update release note during backport.

**Comment by martin-gaievski** [comment]
We may need to add https://github.com/opensearch-project/neural-search/pull/396 for 2.11 release, hope it can be finalized today

**Comment by martin-gaievski** [comment]
> We may need to add #396 for 2.11 release, hope it can be finalized today

We have merged that to 2.11 https://github.com/opensearch-project/neural-search/pull/400, please update notes by including change from that PR

**Review by martin-gaievski — APPROVED** [review]

**Review by junqiu-lei — APPROVED** [review]

**Pull request merged**